web/src/components/datasets/DatasetTags.tsx

```diff
@@ -119,19 +93,37 @@ const DatasetTags: React.FC<IProps> = (props) => {
     setTagDescription(event.target.value)
   }
 
-  const tagData = useSelector((state: IState) => state.tags.tags)
+  const tagData = useSelector((state: IState) =>
+    state.tags.tags.sort((a, b) => a.name.localeCompare(b.name))
+  )
 
-  const handleTagListChange = (event: any) => {
-    setListTag(event.target.value)
-  }
+  const handleTagChange = (
+    _event: React.SyntheticEvent,
+    _value: string[],
+    _reason: AutocompleteChangeReason,
+    details?: AutocompleteChangeDetails<string> | undefined
+  ) => {
+    
+    if (details) {
+      const newTag = details.option
+      const newSelectedTags = [...selectedTags, newTag]
+      setSelectedTags(newSelectedTags)
 
-  const handleTagChange = () => {
-    datasetField
-      ? addDatasetFieldTag(namespace, datasetName, listTag, datasetField)
-      : addDatasetTag(namespace, datasetName, listTag)
+      datasetField
+        ? addDatasetFieldTag(namespace, datasetName, newTag, datasetField)
+        : addDatasetTag(namespace, datasetName, newTag)
+    }
   }
 
   const handleDelete = (deletedTag: string) => {
+    const index = selectedTags.indexOf(deletedTag)
```

Comment:
We can do this a little more functionally like:
```tsx
const newSelectedTags = selectedTags.filter((tag) => deletedTag !== tag);
setSelectedTags(newSelectedTags);
```

Reply:
Maybe let's make this change and I'll merge if that sounds good?

Reply:
Surely can.

Reply:
Could I also do this or does it make it less readable?

```
setSelectedTags(selectedTags.filter((tag) => deletedTag !== tag))
```